core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2118,7 +2139,7 @@ class Log(@volatile private var _dir: File,
    *
    *  @param newOffset The new offset to start the log with
    */
-  private[log] def truncateFullyAndStartAt(newOffset: Long): Unit = {
+  def truncateFullyAndStartAt(newOffset: Long): Unit = {
```

Comment:
It is a little unfortunate to expose the truncation APIs from `Log`. For logs which are managed by `LogManager`, truncation calls are expected to go through `LogManager`. That makes me think it might be useful to have a distinction between the low-level `Log` object and that which is managed by `LogManager`. For example, maybe we could use `ManagedLog` or something like that. Not something to address here, just food for thought in the way of future improvement.